src/ServerDaemon.chpl

```diff
@@ -724,6 +736,30 @@ module ServerDaemon {
                                                                             elapsed));
             this.shutdown(); 
         }
+
+        /* While waiting for a message from client, do checkpointing if desired. */
+        proc startIdleServer() throws {
+          // Do nothing if the checkpointing module was not included.
+          if commandMap.contains("auto_checkpoint") {
+            const msgArgs = new owned MessageArgs();
+            msgArgs.addPayload(b"idle start");
+            // lock here to avoid dependency CheckpointMsg -> ServerDaemons
+            activityMutex.writeEF("autoCheckpoint");
+            defer { activityMutex.readFE(); }
+            // ignore the returned message, for now
+            executeCommand("auto_checkpoint", msgArgs, st);
+          }
+        }
+
+        /* Signal that idling finished. Does NOT wait for activity completion. */
+        proc stopIdleServer() throws {
+          if commandMap.contains("auto_checkpoint") {
+            const msgArgs = new owned MessageArgs();
+            msgArgs.addPayload(b"idle stop");
```

Comment:
I think this is not wired in right now. In general, it is not clear to me why this is necessary. I am probably missing something about asynchrony, `begin` etc.